core/domain/src/main/java/com/nibalk/tasky/core/domain/util/LocalDateExt.kt

```diff
@@ -0,0 +1,67 @@
+package com.nibalk.tasky.core.domain.util
+
+import java.time.Instant
+import java.time.LocalDate
+import java.time.LocalDateTime
+import java.time.LocalTime
+import java.time.ZoneId
+
+fun LocalDate.getSurroundingDays(
+    before: Int = 12,
+    after: Int = 17
+): List<LocalDate> {
+    val startDate = this.minusDays(before.toLong())
+    return (0 until (before + after))
+        .map {
+            startDate.plusDays(it.toLong())
+        }
+        .toList()
+}
+
+// Long -> LocalDate and LocalDateTime
+
+fun Long.toLocalDate(): LocalDate {
+    return Instant
+        .ofEpochMilli(this)
+        .atZone(ZoneId.systemDefault())
```

Comment:
Here you'll get a timezone issue unless the Long timestamp is represented in the local timezone of the user. However, 99% of long timestamps are in UTC timezone, so you'll also need to interpret them as UTC:
```
return Instant
    .ofEpochMilli(this)
    .atZone(ZoneId.of("UTC))
    .withZoneSameInstant(ZoneId.systemDefault()) // This performs the timezone conversion
    .toLocalDate()
```
Same for the remaining calls here